src/client/client.js

```diff
@@ -134,7 +150,7 @@ class _ClientImpl {
     this.store = null;
 
     if (multiplayer) {
```

Comment:
Let's just add a new option in the `multiplayer` argument that allows specifying a custom client. Something like:

```
Client({
  game: TicTacToe,
  multiplayer: {
    server: 'hostname:port',
    clientImpl: <object>,
  }
})
```

That way it's a tiny surgical change and the rest of the interface remains unchanged.

Reply:
update as requested 

Note that this interface does not reflect that serverUrl and socketOpt are parameters for a specific Multiplayer client type